src/Microsoft.AspNetCore.SignalR/Proxies.cs

```diff
@@ -54,12 +55,28 @@ public Task InvokeAsync(string method, params object[] args)
         }
     }
 
+    public class AllClientsExceptProxy<THub> : IClientProxy
+    {
+        private readonly HubLifetimeManager<THub> _lifetimeManager;
+        private readonly List<string> _excludedIds;
+
+        public AllClientsExceptProxy(HubLifetimeManager<THub> lifetimeManager, List<string> excludedIds)
```

Comment:
`IReadOnlyCollection` and `IReadOnlyList` do not have a `Contains()` method, which is essential to the implementation.

And if we care about performance, we must allow passing an `ISet`-based collection.

So, I think it is better to take a (potentially) writable interface that performs well than take a read-only collection that SignalR must first convert to something like `HashSet`.

That interface ends up being `ICollection` because we also want to allow `IList` for convenience. Otherwise, the overload method taking `params string[]` cannot pass its parameter directly.